subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/BaseIncrementalCompilationAfterFailureIntegrationTest.groovy

```diff
@@ -110,7 +201,6 @@ abstract class BaseIncrementalCompilationAfterFailureIntegrationTest extends Abs
 
     def "writes relative paths to source-to-class mapping after incremental compilation"() {
         given:
-        assumeFalse("Source-to-class mapping is not written for CLI compiler", this.class == JavaSourceCliIncrementalCompilationIntegrationTest.class)
```

Comment:
This test was actually moved from other class when I created BaseIncrementalCompilationAfterFailureIntegrationTest.groovy. So if you actually check the condition:
```
this.class == JavaSourceCliIncrementalCompilationIntegrationTest.class
```
it's never true, since `this.class` is never `JavaSourceCliIncrementalCompilationIntegrationTest.class`.